File: problemas/ad-hoc/1091/1091.c
#include <stdio.h>

int main(){
    int K, N, M, X, Y;

    while(scanf("%d", &K)){
        if(!K)  break;

        scanf("%d %d", &N, &M);
        
        for(int i = 0; i < K; ++i){
            scanf("%d %d", &X, &Y);

            X -= N, Y -= M;

            if(X > 0){
                if(Y > 0)       printf("NE\n");
                else if(Y < 0)  printf("SE\n");
                else            printf("divisa\n");
            }else if(X < 0){
                if(Y > 0)       printf("NO\n");
                else if(Y < 0)  printf("SO\n");
                else            printf("divisa\n");
            }else               printf("divisa\n");
        }
    }

    return 0;
}

File: problemas/ad-hoc/1091/1091.go
package main

import (
    "fmt"
)

func main() {
    var k, n, m, x, y int

    for {
        fmt.Scanf("%d", &k)

        if k == 0 {
            break
        }

        fmt.Scanf("%d %d", &n, &m)
        for i := 0; i < k; i++ {
            fmt.Scanf("%d %d", &x, &y)

            x -= n
            y -= m

            if x > 0 && y > 0 {
                fmt.Printf("NE\n")
            } else if x > 0 && y < 0 {
                fmt.Printf("SE\n")
            } else if x < 0 && y > 0 {
                fmt.Printf("NO\n")
            } else if x < 0 && y < 0 {
                fmt.Printf("SO\n")
            } else {
                fmt.Printf("divisa\n")
            }
        }
    }
}
